### PR TITLE
Fix documentation for bin and extraLibs options

### LINKED ISSUE #520: Add description of the bin option.
> why it so happened that the required library to run is not indicated in the list of dependencies.

It might be a *run-time dynamic linking* library used by [`dlopen`](https://man7.org/linux/man-pages/man3/dlopen.3.html) , [`LoadLibrary`](https://docs.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-loadlibraryw) or something equivalent.

Before seeing this post, I had problem with the OpenSSL library. I had to manually copy "libcrypto-1_1-x64.dll" and "libssl-1_1-x64.dll" to the target directory every time. For now, I can use the option "-bin" and "-libDir" like this:

```powershell
> cqt -bin 'path\to\libssl-1_1-x64.dll,myProgram.exe' -libDir 'path\to\the\parent\dir\of\libcrypto-1_1-x64.dll'
```

The library "libssl-1_1-x64.dll" depends on "libcrypto-1_1-x64.dll", so CQtDeployer will copy them together to the target directory. Nice!

However, I never thought I could use the option "-bin" like this and I had no idea why a list can be its value before. This part of the document needs to be improved, in my opinion.

---

And one more thing, the doc said:

> -extraLibs [list,params]
> Sets the mask of the library name for forced copying.
> Example: "-extraLib mySql" - forces to copy all libraries whose names contain mySql to the project folder. This option is case sensitive.

@EndrII explained further:

> As for the **extraLibs** option, this option is intended for forced copying of dependencies from the system environment similar to **deploySystem**.
> https://github.com/QuasarApp/CQtDeployer/issues/394#issuecomment-656678601

It would be better to add the above explanation to the document.

_Originally posted by @pzhlkj6612 in https://github.com/QuasarApp/CQtDeployer/issues/394#issuecomment-774440191_

### PR DESCRIPTION
**Pull request by EndrII**
fix #520

### CODE REVIEW

**Comment by pzhlkj6612** [comment]
I'm rewriting the instruction about the option "-bin". Please wait.

**Comment by EndrII** [comment]
@pzhlkj6612 okay, create pull request into the  task_520 branch and checkout from it.

**Comment by EndrII** [comment]
@pzhlkj6612 
Do you will make any changes to this PR?. If not, please do a review.

File: docs/en/Options.md [hunk 1]
```diff
@@ -75,7 +74,7 @@ cqtdeployer -option1 value1 -option2 list, of, values ​​flag1 flag2 flag3
 |                             | For example -libDir ~/myLib,~/newLibs                           |
 |   -extraLibs [list,params]  | Sets the mask of the library name for forced copying.           |
 |                             | Example: "-extraLib mySql" - forces to copy all libraries whose |
-|                             |names contain mySql to the project folder. This option is case sensitive. |
+|                             |names contain mySql to the project folder. This option is case sensitive. This option will not copy libraries whose names are not mentioned in the list of dependencies (Example: plugins or libs loaded using **dlopen** or **LoadLibrary** functions).|
```
[the remark this answers is not in the document]
Comment (EndrII) [reply docs/en/Options.md:77]: yes you right.

File: docs/en/Options.md [hunk 2]
```diff
@@ -57,8 +57,7 @@ cqtdeployer -option1 value1 -option2 list, of, values ​​flag1 flag2 flag3
 
 | Option                      | Descriptiion                                              |
 |-----------------------------|-----------------------------------------------------------|
-|   -bin [list, params]       | Deployable file or folder.                | 
-|                             | For example -bin ~/my/project/bin/,~/my/project/bin.exe     |
+|   -bin [list,params]       | Deployable file or folder. For example -bin ~/my/project/bin/,~/my/project/bin.exe. The bin option can also work with files of any format. Everything that will be listed as arguments to this option will be copied to the DestributionKit/bin folder and will be located next to the executable files.   |
```
[the remark this answers is not in the document]
Comment (EndrII) [reply docs/en/Options.md:60]: 1. ok i will add description for the deployable files.
2. no, the Distribution Kit have a strong structure, all files marked like bin must be copied in to binOut folder.
3. Yes, you right.

[the remark this answers is not in the document]
Comment (pzhlkj6612) [reply docs/en/Options.md]: > no, the Distribution Kit have a strong structure, all files marked like bin must be copied in to binOut folder.

Well, what if the user use the parameter `targetDir`?

[the remark this answers is not in the document]
Comment (EndrII) [reply docs/en/Options.md]: targetDir sets root for your distribution and XXXoutDir options sets releastive path from root.
example 
targetDir = /opt/MyApp
binDir = targetDir + binPutDir = /opt/MyApp/bin

[the remark this answers is not in the document]
Comment (pzhlkj6612) [reply docs/en/Options.md]: I don't think we should assume that the option "binOutDir" exists by default.

<br>

And,

> Why "DestributionKit/bin"? It is better to use "the destination directory".

I mean that we should not use the hardcoded name "DestributionKit". This name did not exist in Option.md before the modification, right?

[the remark this answers is not in the document]
Comment (EndrII) [reply docs/en/Options.md]: oh i understand you, yes i will remove this hard code from docs.

File: docs/en/Options.md [hunk 3]
```diff
@@ -16,6 +16,14 @@ cqtdeployer -option1 value1 -option2 list, of, values ​​flag1 flag2 flag3
...
 * **clear** - boolean option
 
+## Definitions in description 
+**System environment** - is  pathes of the sstem libraryes.
+    Windows: C:\Windows\System32 and C:\Windows\SysWOW64. And any another pathes difined in the PATH variable.
+    Linux: /usr/lib and /lib. And any another pathes defined in the LD_LIBRARY_PATH or PATH variables.
+
+**Deployable files** - is all file that will be copy in to targer dirictory. (libraries, resources)
+**Targets** - is all files marked by bin option.
+
```
Comment (EndrII) [inline docs/en/Options.md:26]:
```suggestion
## Definitions in description 
**System environment** - is  pathes of the sstem libraryes.
    Windows: C:\Windows\System32 and C:\Windows\SysWOW64. And any another pathes difined in the PATH variable.
    Linux: /usr/lib and /lib. And any another pathes defined in the LD_LIBRARY_PATH or PATH variables.

**deployable files** - are all file that will be copied into distribution (incude executable files)

**Targets** - are all executable files marked by bin option.
**Executable files ** - are all files with PE, ELF and Mach-O formats.

```

**Review by pzhlkj6612 — CHANGES_REQUESTED** [review]

File: Deploy/deploycore.cpp [hunk 4]
```diff
@@ -225,7 +225,14 @@ void DeployCore::help() {
...
             "Part 2 Deploy options", {
-                {"-bin [list, params]", "Deployable file or folder. For example -bin ~/my/project/bin/,~/my/project/bin.exe"},
+                {"-bin [list, params]", "Files to deploy or folders that contain files to deploy."
+                 " For example -bin ~/my/project/bin/,~/my/project/bin.exe,~/my/project/runtimeLinking/lib.dll."
+                 " For files: These files will be unconditional copied to the destination directory,"
+                 " regardless of their format or suffix."
+                 " For folders:"
+                 " CQtDeployer will enter these folders and non-recursively copy all deployable files to the destination directory."
+                 " Then, CQtDeployer will be extract all dependencies of the copied files."
+                 " The CQtDeployer will be search dependencies in a system environment and in libDir paths."},
```
Comment (EndrII) [inline Deploy/deploycore.cpp:235]:
```suggestion
                 " For folders:"
                 " CCQtDeployer will enter these folders and non-recursively copy all executable files to the destination directory."
                 " Then, CQtDeployer will extract all dependencies of the copied files and search dependencies in system environments and libDir paths."},
```

File: docs/en/Options.md [hunk 5]
```diff
@@ -57,7 +65,7 @@ cqtdeployer -option1 value1 -option2 list, of, values ​​flag1 flag2 flag3
...
+|   -bin [list,params]       | Files to deploy or folders that contain files to deploy. For example -bin ~/my/project/bin/,~/my/project/bin.exe,~/my/project/runtimeLinking/lib.dll. For files: These files will be unconditional copied to the destination directory, regardless of their format or suffix. For folders: CQtDeployer will enter these folders and non-recursively copy all deployable files to the destination directory. Then, CQtDeployer will be extract all dependencies of the copied files. The CQtDeployer will be search dependencies in a system environment and in libDir paths. |
```
[the remark this answers is not in the document]
Comment (EndrII) [reply docs/en/Options.md:68]: i am merge it

File: Deploy/deploycore.cpp (hunk 4 above)
Comment (pzhlkj6612) [reply Deploy/deploycore.cpp:235]: Oh I forgot this.

File: docs/en/Options.md (hunk 3 above)
Comment (EndrII) [reply docs/en/Options.md:26]: @pzhlkj6612 What do you think about this ?

File: docs/ru/Options.md [hunk 6]
```diff
@@ -18,6 +18,14 @@ cqtdeployer -option1 value1 -option2 list,of,values flag1 flag2 flag3
...
  * **clear** - флаг
 
+## Определения в описании
+**Системное окружение** - пути к системным библиотекам.
+     Windows: C:\Windows\System32 и C:\Windows\SysWOW64. И любые другие пути, указанные в переменной PATH.
+     Linux: /usr/lib и /lib. И любые другие пути, определенные в переменных LD_LIBRARY_PATH или PATH.
+
+**Развертываемые файлы** - это все файлы, которые будут скопированы в каталог назначения. (библиотеки, ресурсы)
+**Цели** - это все файлы, отмеченные опцией bin. 
+
```
Comment (EndrII) [inline docs/ru/Options.md:28]:
```suggestion
## Определения в описании
**Системное окружение** - пути к системным библиотекам.
  * Windows:
    * Пути, определенные в переменной PATH.
    * C:\Windows\System32
    * C:\Windows\SysWOW64
  * Linux:
    * Пути, определенные в переменных LD_LIBRARY_PATH и PATH.
    * /usr/lib
    * /lib
**Развертываемые файлы** - это все файлы, которые будут скопированы в дистрибутив (включая исполняемые файлы)
**Цели** - это все исполняемые файлы, отмеченные опцией bin.
**Исполняемые файлы** - это все файлы в форматах PE, ELF и Mach-O.
```

**Review by pzhlkj6612 — CHANGES_REQUESTED** [review]

**Review by pzhlkj6612 — APPROVED** [review]
The documentation for `bin` and `extraLibs` options in this PR looks good to me, so merge it.

**Pull request merged**